### PR TITLE
Change incorrect 'optional' argument option to 'required'

### PR DESCRIPTION
**Pull request by othercorey**
Refs https://github.com/cakephp/cakephp/issues/16158

### CODE REVIEW

File: src/Console/ConsoleInputArgument.php [hunk 1]
```diff
@@ -25,6 +26,7 @@
  *
  * @see \Cake\Console\ConsoleOptionParser::addArgument()
  */
+#[AllowDynamicProperties]
```
Comment (dereuromark) [inline src/Console/ConsoleInputArgument.php]: Should be fqcn inline

**Comment by saeideng** [comment]
we no need to dynamic property in this class
only needs to check valid property before assignment

**Comment by markstory** [comment]
Are these dynamic properties? All the possible options should be defined as properties on the class already.

**Comment by othercorey** [comment]
> Are these dynamic properties? All the possible options should be defined as properties on the class already.

```
68) Cake\Test\TestCase\Console\CommandRunnerTest::testRunValidCommandInflection
Creation of dynamic property Cake\Console\ConsoleInputArgument::$_optional is deprecated

/home/runner/work/cakephp/cakephp/src/Console/ConsoleInputArgument.php:70
/home/runner/work/cakephp/cakephp/src/Console/ConsoleOptionParser.php:499
/home/runner/work/cakephp/cakephp/src/Command/SchemacacheBuildCommand.php:87
/home/runner/work/cakephp/cakephp/src/Console/BaseCommand.php:113
/home/runner/work/cakephp/cakephp/src/Console/BaseCommand.php:154
/home/runner/work/cakephp/cakephp/src/Console/CommandRunner.php:334
/home/runner/work/cakephp/cakephp/src/Console/CommandRunner.php:172
/home/runner/work/cakephp/cakephp/tests/TestCase/Console/CommandRunnerTest.php:276
```

**Comment by saeideng** [comment]
>Creation of dynamic property Cake\Console\ConsoleInputArgument::$_optional is deprecated

this is because of wrong use of `addArgument()`
```
->addArgument('name', [
            'help' => '',
            'optional' => true,
        ])
```

`optional` should remove or replace with `'required' => false`

**Comment by othercorey** [comment]
Looks like we have some broken tests.

**Comment by othercorey** [comment]
Changed to fix the incorrect `optional` options.

**Comment by ADmad** [comment]
If the option is incorrect the fix should be done in 4.x itself.

**Comment by othercorey** [comment]
Sure. Technically, neither the old or new code makes a difference in runtime.

**Comment by ADmad** [comment]
Given that this incorrect option wasn't detected earlier it would be nice to have a test which exercises the `required` option.

**Comment by othercorey** [comment]
The testRunOptionParserFailure() test in CommandTest.php does test this.

**Review by ADmad — APPROVED** [review]

**Pull request merged**